### PR TITLE
Add Kubeconfig Token

### PR DESCRIPTION
**Pull request by devdua**
This PR adds the brand new Kubeconfig Canarytoken

### CODE REVIEW

File: kubeconfig.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+            kc['current-context'] = "%s-%s" % (username, cluster_name)
+
+            # Custom representer to make OrderDict parseable by pyyaml
+            preserve_order = lambda self, data:  self.represent_mapping('tag:yaml.org,2002:map', data.items())
+            yaml.add_representer(OrderedDict, preserve_order)
+            # 0: Truncated cert fingerprint, 1: b64 encoded kubeconfig
+            return (client_auth["f"].replace(":","")[:25].lower(), base64.b64encode(yaml.dump(kc, None, default_flow_style=False, sort_keys=False)))
+        except Exception as e:
+            log.error(u"%s" % e)
+            print e
```
Comment (jayjb) [inline kubeconfig.py:94]: We already logging the error; we can remove the print

File: queries.py [hunk 2]
```diff
@@ -563,6 +563,33 @@ def update_tor_exit_nodes_loop():
...
+    hit_key = "kchit:{}".format(key)
+    db.hset(hit_key, 'hits', hits)
+
+    if not update:
+        # typical timeout sent with each kubectl caching discovery request is 32s, and 5 requests are sent as part of each kubectl execution
+        db.expire(hit_key, 5*32)
+
+def get_kc_hits(key):
+    return (db.hgetall("kchit:{}".format(key)), db.pttl("kchit:{}".format(key)))
+
```
Comment (jayjb) [inline queries.py:592]: Would you mind moving the redis keys to `redismanager.py`? We try keep a listing of the used keys there

**Review by jayjb — CHANGES_REQUESTED** [review]
Thanks @devdua. Let me know what you think

File: kubeconfig.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
+from twisted.internet import ssl
```
[the remark this answers is not in the document]
Comment (devdua) [reply kubeconfig.py:1]: Thanks, removed!

File: channel_input_mtls.py [hunk 4]
```diff
@@ -0,0 +1,331 @@
+from OpenSSL.crypto import FILETYPE_PEM, PKey, TYPE_RSA, X509, X509Extension, dump_certificate, dump_privatekey, load_certificate, load_privatekey
+from twisted.python.filepath import FilePath
```
[the remark this answers is not in the document]
Comment (devdua) [reply channel_input_mtls.py:2]: Thanks, removed all except `PrivateCertificate`, which is used.

File: channel_input_mtls.py [hunk 5]
```diff
@@ -0,0 +1,331 @@
...
+        if trigger['dispatch']:
+            dispatcher(
+                canarydrop=canarydrop,
+                src_ip=trigger['ip'],
+                useragent=trigger['useragent'],
+                location=path,
+                additional_info=trigger['additional_info'])
+
+    def _get_ssl_context(self):
+        from twisted.python.filepath import FilePath
```
[the remark this answers is not in the document]
Comment (devdua) [reply channel_input_mtls.py:289]: Done

File: channel_input_mtls.py [hunk 6]
```diff
@@ -0,0 +1,331 @@
...
+        self.bodies = bodies
+        self.client_ca_cert_path = ca_cert_path
+        self.enricher = enricher
+
+        self.switchboard = kw.pop('switchboard')
+        InputChannel.__init__(self, switchboard=self.switchboard,
+                              name=channel_name if channel_name is not None else self.CHANNEL,
+                              unique_channel=False)
+
+    def buildProtocol(self, addr):
```
[the remark this answers is not in the document]
Comment (devdua) [reply channel_input_mtls.py:198]: It's a required arg as per https://twistedmatrix.com/documents/19.10.0/api/twisted.internet.protocol.Factory.html#buildProtocol

File: queries.py (hunk 2 above)
Comment (devdua) [reply queries.py:592]: Done!

**Review by jayjb — APPROVED** [review]

**Pull request merged**